Main.hs

```diff
@@ -425,7 +425,7 @@ compileInstRule cs ty = case unSpine  $ ty of
     where dropPi = underAbstr a b (compileInstRule cs . unEl)
   _ -> __IMPOSSIBLE__
 
-compileInstanceClause :: LocalDecls -> Clause -> TCM (Hs.InstDecl ())
+compileInstanceClause :: LocalDecls -> Clause -> TCM [Hs.InstDecl ()]
```

Comment:
Since we are returning either 0 or 1 declarations (depending on the visibility of the argument), I would prefer returning `TCM (Maybe (Hs.InstDecl ()))`, and then using `catMaybe` instead of `concat` in `compileInstance`.